src/util-inl.h

```diff
@@ -562,6 +562,22 @@ constexpr std::string_view FastStringKey::as_string_view() const {
   return name_;
 }
 
+// Converts a V8 numeric value to a corresponding C++ primitive or enum type.
+template <typename T>
+T FromV8Value(v8::Local<v8::Value> value) {
+  CHECK(std::is_arithmetic<T>::value || std::is_enum<T>::value);
+  CHECK(value->IsNumber());
+  if constexpr (std::is_integral<T>::value && std::is_unsigned<T>::value) {
+    return static_cast<T>(value.As<v8::Uint32>()->Value());
+  } else if constexpr ((std::is_integral<T>::value &&
+                        std::is_signed<T>::value) ||
+                       std::is_enum<T>::value) {
+    return static_cast<T>(value.As<v8::Int32>()->Value());
```

Comment:
These should also compare `std::numeric_limits<T>` with `std::numeric_limits<uint32_t>`/`std::numeric_limits<int32_t>`.